service/api_task.go

```diff
@@ -305,6 +328,22 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, taskQueueService model.
 		}
 	}
 
+	// STU: Can we just get the distro from the database once, right here? Is there a need for us to go to the database to get the distro on every iteration for the loop below?
+	// STU: should we bailout if there is a database error leaving us unsure if the distro document actual exists?
+	// d, err := distro.FindOne(distro.ById(currentHost.Distro.Id))
```

Comment:
I think we do need to get it from the db in order to ensure that we correctly handle distros that have been disabled/etc.

Reply:
As clarified 1:1 -- prior to entering the loop we'll only go to the database once to get the `distro` and fallback on `currentHost.Distro` is necessary.  I believe there is no benefit to be had by repeating this process on every iteration of the loop like we currently do.